src/anbox/cmds/launch.cpp

```diff
@@ -69,51 +86,95 @@ anbox::cmds::Launch::Launch()
     auto bus = std::make_shared<core::dbus::Bus>(core::dbus::WellKnownBus::session);
     bus->install_executor(core::dbus::asio::make_executor(bus, rt->service()));
 
+    const auto snap_path = utils::get_env_value("SNAP");
+    if (!snap_path.empty()) {
+      const auto resource_path = utils::string_format("%s/usr/share/anbox", snap_path);
+      SystemConfiguration::instance().set_resource_path(resource_path);
+    }
+
+    std::shared_ptr<ui::SplashScreen> ss;
+
+    // Instead of relying on the user session init system to start our
+    // session manager process we also attempt to start it on our own
+    // if not already running. This will help to mitigate problems with
+    // a crashing or a not yet started session manager instance.
     std::shared_ptr<dbus::stub::ApplicationManager> stub;
-    try {
-      stub = dbus::stub::ApplicationManager::create_for_bus(bus);
-    } catch (...) {
-      ERROR("Anbox session manager service isn't running!");
-      return EXIT_FAILURE;
+    for (auto n = 0; n < max_restart_attempts; n++) {
+      try {
+        stub = dbus::stub::ApplicationManager::create_for_bus(bus);
+        break;
+      } catch (std::exception &err) {
+        WARNING("Anbox session manager service isn't running, trying to start it.");
+
+        // Give us a splash screen as long as we're trying to connect
+        // with the session manager so the user knows something is
+        // happening after he started Anbox.
+        if (!ss)
+          ss = std::make_shared<ui::SplashScreen>();
+
+        std::vector<std::string> args = {"session-manager"};
+
+        std::map<std::string,std::string> env;
+        core::posix::this_process::env::for_each([&](const std::string &name, const std::string &value) {
+          env.insert({name, value});
+        });
+
+        const auto exe_path = utils::process_get_exe_path(::getpid());
+        if (!fs::exists(exe_path)) {
+          ERROR("Can't find correct anbox executable to run. Found %s but does not exist", exe_path);
+          return EXIT_FAILURE;
+        }
+
+        try {
+          const auto flags = core::posix::StandardStream::empty; // core::posix::StandardStream::stdout | core::posix::StandardStream::stderr;
```

Comment:
@vosst I didn't looked into the implementation of process-cpp yet but `empty` seems to do what I would expect from `stdoud | stderr`: Allowing the child to print everything on the parent stdout/stderr.